### DISCUSSION TITLE
Next.JS 12.2 Middleware redirect not loading page

### DISCUSSION THREAD

**Discussion by rlawrence89** [vercel/next.js#38216] [Help] [answered]
Here is my second issue after upgrading to next 12.2. I have moved my middleware file to the appropriate directory and performed the necessary updates. Now my app doees not load the login page. 

Here is the piece of code that is the culprit.

```javascript
f (!token && pathname !== 'http://localhost:3000/login') {
      console.log('Redirecting to login')
      const loginUrl = new URL('/login', req.url)
      return NextResponse.redirect(loginUrl)
}
```

If i comment out that line my login page loads. But if a user just visits www.localhost.com it will not redirect to login.

Currently using NExt-Auth with the credentials provider.

**Comment by icyJoseph** [comment, marked as answer]
Hi,

The following snippet works:

```ts
import { NextResponse, type NextRequest } from "next/server";

export function middleware(req: NextRequest) {
  const { pathname } = req.nextUrl;

  if (pathname.startsWith("/_next")) return NextResponse.next();

  if (pathname === "/") {
    req.nextUrl.pathname = "/blog";
    return NextResponse.redirect(req.nextUrl);
  }

  return NextResponse.next();
}
```

Where you think we diverge from each other? `pathname` doesn't include the origin IIRC, and you don't need to make a new URL, IIRC.

**Reply by rlawrence89** [reply]
Thank you. must have missed that "_next". Its got to be new because it was working before the update.

**Reply by icyJoseph** [reply]
Ah yeah they changed that in the new update. All Next.js internals are also seen by middleware. https://nextjs.org/docs/messages/middleware-upgrade-guide#executing-middleware-on-internal-nextjs-requests

**Reply by Gourav-vats** [reply]
It worked! but on the login page i was using an nextImage component and since we are just doing NextResponse.next() whenever it goes to /_next it is not able to use nextImage optimization. What should do ?

**Reply by skychavda** [reply]
Can you check this I am trying to redirect the user to a new route but it didn't work. https://github.com/vercel/next.js/issues/59218#issuecomment-2217270733
